main/realmain.cc

```diff
@@ -104,6 +104,22 @@ core::StrictLevel levelMinusOne(core::StrictLevel level) {
     }
 }
 
+// Filter levels to a sensible recommendation.
+core::StrictLevel levelToRecommendation(core::StrictLevel level) {
+    switch (level) {
+        case core::StrictLevel::Ignore:
+        case core::StrictLevel::False:
+        case core::StrictLevel::True:
+        case core::StrictLevel::Strict:
+            return level;
+        case core::StrictLevel::Strong:
+        case core::StrictLevel::Max:
+            return core::StrictLevel::Strict;
+        default:
+            Exception::raise("Should never happen");
```

Comment:
If you delete the `default` case, clang will check the exhaustiveness for you.

Reply:
I gave this a try, and the remaining StrictLevel values were `Internal`, `None`, `Autogenerated`, and `StdLib`. Should we just map them back to themselves for the purposes of recommendation?

Reply:
I made those cases explitic, and raised the exception that was in the `default` case before.